### DISCUSSION TITLE
S5 - mqtt error on 'custom' fan speed

### DISCUSSION THREAD

**Discussion by rjlee** [Hypfer/Valetudo#1807] [Q&A / Support] [answered]
Hi,

I've recently installed valetudo on an imported S5, all works fine, however I see the logs spammed by:

`Failed to parse value to be sent for topic valetudo/myvac/FanSpeedControlCapability/preset: Error: Value 'custom' is invalid for enum with format [low,medium,high,max,off]!
    at PropertyMqttHandle.marshalHomie (/snapshot/Valetudo/backend/lib/mqtt/handles/MqttHandle.js:371:27)
    at PropertyMqttHandle.getHomie (/snapshot/Valetudo/backend/lib/mqtt/handles/MqttHandle.js:298:29)
    at async MqttController.refresh (/snapshot/Valetudo/backend/lib/mqtt/MqttController.js:714:23)
    at async PropertyMqttHandle.refresh (/snapshot/Valetudo/backend/lib/mqtt/handles/MqttHandle.js:237:9)
    at async PresetSelectionCapabilityMqttHandle.refresh (/snapshot/Valetudo/backend/lib/mqtt/handles/MqttHandle.js:240:13)`

In valetudo I see the fan is set as 'custom':

![image](https://user-images.githubusercontent.com/18695/233863959-dc40cfe0-9579-46fc-aab3-23b68a0cd04b.png)

Setting this to a 'valid' value stops the log spam, but whenever the rebot reboots it always is reset to 'custom'. 

Is this a known issue/any suggested fixes ?

Thanks

**Comment by Hypfer** [comment]
Huh, interesting. Is it running the latest firmware? I _think_ i recall fanspeeds changing at some point in its version history

**Reply by rjlee** [reply]
Yes it is.

![image](https://user-images.githubusercontent.com/18695/234090473-19b922d9-9f11-4479-9f9b-7860737bffd1.jpeg)

**Reply by Hypfer** [reply, marked as answer]
No it's not :)
![image](https://user-images.githubusercontent.com/974410/234095622-cff61892-a978-4e03-b842-3a07089f5af2.png)

You likely want to run the latest firmware that is 2034
https://builder.dontvacuum.me/_s5.html

See also:
https://valetudo.cloud/pages/usage/firmware-updates.html

**Reply by rjlee** [reply]
Ah sorry, I misread the question !  I'm on the latest Valetudo ...  I'll have a go at upgrading the firmware as you suggest and report back, hopefully that addresses the issue 👍

**Comment by rjlee** [comment]
To confirm, that fixes the issue, many thanks.
